core/domain/blog_services.py

```diff
@@ -0,0 +1,613 @@
+        blog_post_summary_model.author_id,
+        blog_post_summary_model.title,
+        blog_post_summary_model.summary,
+        blog_post_summary_model.url_fragment,
+        blog_post_summary_model.tags,
+        blog_post_summary_model.thumbnail_filename,
+        blog_post_summary_model.last_updated,
+        blog_post_summary_model.published_on)
+
+
+def get_blog_post_summary_by_id(blog_post_id, strict=False):
+    """Returns a domain object representing a blog post summary.
+
+    Args:
+        blog_post_id: str. ID of the blog post.
+        strict: bool. Fails noisily if the model doesn't exist.
+
+    Returns:
+        BlogPostSummary or None. The domain object representing a blog post
+        summary with the given ID, or None if it does not exist.
+    """
+    blog_post_summary_model = blog_models.BlogPostSummaryModel.get(
+        blog_post_id, strict=strict)
+    if blog_post_summary_model:
+        blog_post_summary = get_blog_post_summary_from_model(
+            blog_post_summary_model)
+        return blog_post_summary
+    else:
+        return None
+
+
+def get_blog_post_summary_models_list_by_user_id(
+        user_id, blog_post_is_published):
+    """Given the user ID and status, it returns the list of blog post summary
+    domain object for which user is an editor and the status matches.
+
+    Args:
+        user_id: str. The user who is editor of the blog posts.
+        blog_post_is_published: bool. Whether the given blog post is
+            published or not.
+
+    Returns:
+        list(BlogPostSummary). The blog post summaries of the blog posts for
+        which the user is an editor corresponding to the status
+        (draft/published).
+    """
+    blog_post_ids = filter_blog_post_ids(user_id, blog_post_is_published)
+    blog_post_summary_models = (
+        blog_models.BlogPostSummaryModel.get_multi(blog_post_ids))
+    blog_post_summaries = []
+    blog_post_summaries = [
+        get_blog_post_summary_from_model(model) if model is not None else None
+        for model in blog_post_summary_models]
+    return blog_post_summaries if len(blog_post_summaries) != 0 else None
+
+
+def filter_blog_post_ids(user_id, blog_post_is_published):
+    """Given the user ID and status, it returns the IDs of all blog post
+    according to the status.
+
+    Args:
+        user_id: str. The user who is editor of the blog post.
+        blog_post_is_published: bool. True if blog post is published.
+
+    Returns:
+        list(str). The blog post IDs of the blog posts for which the user is an
+        editor corresponding to the status(draft/published).
+    """
+    blog_post_rights_models = (
+        blog_models.BlogPostRightsModel.get_by_user(user_id))
+    model_ids = []
+    if blog_post_rights_models:
+        for model in blog_post_rights_models:
+            if model.blog_post_is_published == blog_post_is_published:
+                model_ids.append(model.id)
+    return model_ids
```

Comment:
Why are we looping over all the models? Instead, can we query over author_id &  blog_post_is_published? [Note: `get_by_user` will return a limited number of models]

Reply:
Done.